src/charm.py

```diff
@@ -531,12 +531,16 @@ def _update_kratos_endpoints_relation_data(self, event: RelationEvent) -> None:
         logger.info("Sending endpoints info")
 
         admin_endpoint = (
-            f"http://{self.app.name}.{self.model.name}.svc.cluster.local:{KRATOS_ADMIN_PORT}"
+            self.admin_ingress.url
+            if self.admin_ingress.is_ready()
+            else f"http://{self.app.name}.{self.model.name}.svc.cluster.local:{KRATOS_ADMIN_PORT}",
         )
         public_endpoint = (
-            f"http://{self.app.name}.{self.model.name}.svc.cluster.local:{KRATOS_PUBLIC_PORT}"
+            self.public_ingress.url
+            if self.public_ingress.is_ready()
+            else f"http://{self.app.name}.{self.model.name}.svc.cluster.local:{KRATOS_PUBLIC_PORT}",
         )
-        self.endpoints_provider.send_endpoint_relation_data(admin_endpoint, public_endpoint)
+        self.endpoints_provider.send_endpoint_relation_data(admin_endpoint[0], public_endpoint[0])
```

Comment:
Why do we need to return `[0]`?

Reply:
It's because this returns a tuple:
```python3
public_endpoint = (
	self.public_ingress.url
	if self.public_ingress.is_ready()
	else f"http://{self.app.name}.{self.model.name}.svc.cluster.local:{KRATOS_PUBLIC_PORT}",
    )
```
I can cast it to string though, so that we don't have to return `[0]`.

Reply:
ohhh, you need to remove the trailing comma